edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/metrics/MetricsCommandFactory.cs

```diff
@@ -29,6 +29,15 @@ public async Task<ICommand> CreateAsync(IModuleWithIdentity module, IRuntimeInfo
             }
         }
 
+        public async Task<ICommand> PrepareUpdateAsync(IModule module, IRuntimeInfo runtimeInfo)
+        {
+            this.factoryMetrics.AddMessage(module, FactoryMetrics.ModuleCommandMetric.PrepareUpdate);
+            using (this.factoryMetrics.MeasureTime("prepareUpdate"))
+            {
+                return await this.underlying.PrepareUpdateAsync(module, runtimeInfo);
+            }
+        }
```

Comment:
Want to call out this metric is being added. I need to figure out if we need to update ValidateMetrics E2E test. We already updated the docs BuiltInMetrics.md.

Reply:
Based on existing pattern in ValidateMetrics E2E test I don't `command_latency_seconds` being tested

Reply:
Actually this is being tested, will add appropriate tests.